gui/default/syncthing/settings/advancedSettingsModalView.html

```diff
@@ -78,10 +78,10 @@ <h4 ng-if="folder.label.length != 0" class="panel-title" tabindex="0">
           <div class="panel-body">
             <form class="form-horizontal" role="form">
               <div ng-repeat="(key, value) in folder" ng-if="inputTypeFor(key, value) != 'skip'" class="form-group">
-                <label for="folder{{$index}}Input{{$index}}" class="col-sm-4 control-label">{{key | uncamel}}</label>
+                <label for="folder{{$parent.$parent.$index}}Input{{$index}}" class="col-sm-4 control-label">{{key | uncamel}}</label>
                 <div class="col-sm-8">
-                  <input ng-if="inputTypeFor(key, value) == 'list'" id="optionsInput{{$index}}" class="form-control" type="text" ng-model="folder[key]" ng-list />
-                  <input ng-if="inputTypeFor(key, value) != 'list'" id="optionsInput{{$index}}" class="form-control" type="{{inputTypeFor(key, value)}}" ng-model="folder[key]" />
+                  <input ng-if="inputTypeFor(key, value) == 'list'" id="folder{{$parent.$parent.$parent.$index}}Input{{$index}}" class="form-control" type="text" ng-model="folder[key]" ng-list />
```

Comment:
The `$parent.$parent.$parent` stuff looks rather fragile. Maybe we can use the folder ID / device ID in these places if the counter index isn't easily available?

Reply:
You can use `ng-init` to give the index a name so you can access it here: https://code.angularjs.org/1.3.20/docs/api/ng/directive/ngInit

Reply:
I have tried to modify the code using the `ng-init` approach. Please check and let me know what you think about the current iteration.